app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -537,6 +544,18 @@ public void held(StreamInfoItem selectedItem) {
         detailControlsDownload.setOnClickListener(this);
         detailControlsDownload.setOnLongClickListener(this);
         relatedStreamExpandButton.setOnClickListener(this);
+        switchAutoplay.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
+                SharedPreferences autoplaysetting = PreferenceManager.getDefaultSharedPreferences(activity);
+                if(switchAutoplay.isChecked()){
```

Comment:
It may be simple
```              
PreferenceManager.getDefaultSharedPreferences(activity).edit()
    .putBoolean(getString(R.string.auto_queue_key),switchAutoplay.isChecked())
    .apply();
```

Reply:
Thank you both for you replies! I commited the simple version as you wished. The previouse one, maybe was more understandable, but yes it can be simple to - no big difference ;)